app/test/unit/git/cherry-pick-test.ts

```diff
@@ -264,4 +271,126 @@ describe('git/cherry-pick', () => {
+    })
+
+    it('successfully continue cherry pick with conflicts after resolving them', async () => {
+      result = await cherryPick(repository, featureBranch.tip.sha)
+      expect(result).toBe(CherryPickResult.ConflictsEncountered)
+
+      const statusAfterCherryPick = await getStatusOrThrow(repository)
+      const { files } = statusAfterCherryPick.workingDirectory
+
+      const diffCheckBefore = await GitProcess.exec(
+        ['diff', '--check'],
+        repository.path
+      )
+
+      expect(diffCheckBefore.exitCode).toBeGreaterThan(0)
+
+      // resolve conflicts by writing files to disk
+      await FSE.writeFile(
+        Path.join(repository.path, 'THING.md'),
+        '# HELLO WORLD! \nTHINGS GO HERE\nFEATURE BRANCH UNDERWAY\n'
+      )
+
+      const diffCheckAfter = await GitProcess.exec(
+        ['diff', '--check'],
+        repository.path
+      )
+
+      expect(diffCheckAfter.exitCode).toEqual(0)
+
+      result = await continueCherryPick(repository, files)
+
+      expect(result).toBe(CherryPickResult.CompletedWithoutError)
+    })
+
+    it('successfully detect cherry pick with outstanding files not staged', async () => {
+      result = await cherryPick(repository, featureBranch.tip.sha)
+      expect(result).toBe(CherryPickResult.ConflictsEncountered)
+
+      result = await continueCherryPick(repository, [])
+      expect(result).toBe(CherryPickResult.OutstandingFilesNotStaged)
+
+      const status = await getStatusOrThrow(repository)
+      const conflictedFiles = status.workingDirectory.files.filter(
+        f => f.status.kind === AppFileStatusKind.Conflicted
+      )
+      expect(conflictedFiles).toHaveLength(1)
+    })
+
+    it('successfully continue cherry pick with additional changes unrelated to conflicted files', async () => {
+      result = await cherryPick(repository, featureBranch.tip.sha)
+      expect(result).toBe(CherryPickResult.ConflictsEncountered)
+
+      // resolve conflicts by writing files to disk
+      await FSE.writeFile(
+        Path.join(repository.path, 'THING.md'),
+        '# HELLO WORLD! \nTHINGS GO HERE\nFEATURE BRANCH UNDERWAY\n'
+      )
+
+      // change files unrelated to conflicts
+      await FSE.writeFile(
+        Path.join(repository.path, 'UNRELATED_FILE.md'),
+        '# HELLO WORLD! \nUNRELATED FILE STUFF IN HERE\n'
+      )
+
+      const statusAfterCherryPick = await getStatusOrThrow(repository)
+      const { files } = statusAfterCherryPick.workingDirectory
+
+      // THING.MD and UNRELEATED_FILE.md should be in working directory
+      expect(files.length).toBe(2)
+
+      result = await continueCherryPick(repository, files)
+      expect(result).toBe(CherryPickResult.CompletedWithoutError)
+
+      // Only UNRELATED_FILE.md should be in working directory
+      // THING.md committed with cherry pick
+      const status = await getStatusOrThrow(repository)
+      expect(status.workingDirectory.files[0].path).toBe('UNRELATED_FILE.md')
+    })
```

Comment:
Instead of saying "files unrelated to conflicts", maybe we should be more specific and say untracked files?

I ask because you could change an existing tracked file, and that one would be included in the cherry pick, right? What this test is checking only affects untracked files.

```suggestion
    it('successfully continue cherry pick with new untracked files', async () => {
      result = await cherryPick(repository, featureBranch.tip.sha)
      expect(result).toBe(CherryPickResult.ConflictsEncountered)

      // resolve conflicts by writing files to disk
      await FSE.writeFile(
        Path.join(repository.path, 'THING.md'),
        '# HELLO WORLD! \nTHINGS GO HERE\nFEATURE BRANCH UNDERWAY\n'
      )

      // add new untracked file
      await FSE.writeFile(
        Path.join(repository.path, 'UNTRACKED_FILE.md'),
        '# HELLO WORLD! \nUNTRACKED FILE STUFF IN HERE\n'
      )

      const statusAfterCherryPick = await getStatusOrThrow(repository)
      const { files } = statusAfterCherryPick.workingDirectory

      // THING.MD and UNTRACKED_FILE.md should be in working directory
      expect(files.length).toBe(2)

      result = await continueCherryPick(repository, files)
      expect(result).toBe(CherryPickResult.CompletedWithoutError)

      // Only UNTRACKED_FILE.md should be in working directory
      // THING.md committed with cherry pick
      const status = await getStatusOrThrow(repository)
      expect(status.workingDirectory.files[0].path).toBe('UNTRACKED_FILE.md')
    })
```

Reply:
Makes sense -> The term untracked files is a little foreign to me (still coming up to speed with official git terminology). Thus, I was just using words that I was familiar with. 😄